src/aggregator_contract.cairo

```diff
@@ -26,20 +26,45 @@ pub mod AggregatorContract {
             IOwnableContractDispatcher { contract_address: ownable_contract_address }.get_owner()
         }
 
-        fn set_new_ownable_contract_owner(ref self: ContractState, ownable_contract_address : ContractAddress, new_owner: ContractAddress) {
+        fn get_kill_switch_status(
+            self: @ContractState, kill_switch_contract_address: ContractAddress
+        ) -> bool {
+            IKillSwitchDispatcher { contract_address: kill_switch_contract_address }.is_active()
+        }
+
+        fn set_new_ownable_contract_owner(
+            ref self: ContractState,
+            ownable_contract_address: ContractAddress,
+            new_owner: ContractAddress
+        ) {
             assert(new_owner.is_non_zero(), hello_cairo::errors::Errors::ZERO_ADDRESS_OWNER);
-            IOwnableContractDispatcher { contract_address: ownable_contract_address  }.set_owner(new_owner);
+            IOwnableContractDispatcher { contract_address: ownable_contract_address }
+                .set_owner(new_owner);
         }
 
-        fn increase_aggr_count_by_two(ref self: ContractState, counter_contract_adress: ContractAddress) {
-          let current_count =  ICounterDispatcher { contract_address: counter_contract_adress }.get_count();
-          self.aggr_count.write(current_count * 2);
+        fn increase_aggr_count_by_two(
+            ref self: ContractState,
+            counter_contract_adress: ContractAddress,
+            kill_switch_contract_address: ContractAddress
+        ) {
+            let kill_switch_status = IKillSwitchDispatcher {
+                contract_address: kill_switch_contract_address
+            }
+                .is_active();
+            assert(kill_switch_status == true, hello_cairo::errors::Errors::NOT_ACTIVE);
+            let current_count = ICounterDispatcher { contract_address: counter_contract_adress }
+                .get_count();
+            self.aggr_count.write(current_count * 2);
         }
```

Comment:
When you call `increase_aggr_count_by_two`, make sure you utilize the `activate` function of `KillSwitch` to manually turn  `is_active` variable  of your `KillSwitch` contract to true directly within `AggregatorContract`